cmd/tnf/addclaim/addclaim.go

```diff
@@ -27,12 +29,239 @@ var (
 		Short: "The test suite generates a \"claim\" file",
 		RunE:  claimUpdate,
 	}
+	claimCompareFiles = &cobra.Command{
+		Use:   "compare",
+		Short: "Compare 2 \"claim\" file",
+		RunE:  claimCompare,
+	}
 )
 
 const (
 	claimFilePermissions = 0o644
 )
 
+type claimFileStruct struct {
+	Claim struct {
+		Nodes struct {
+			CniPlugins  map[string]cnistruct   `json:"cniPlugins"`
+			NodesHwInfo map[string]interface{} `json:"nodesHwInfo"`
+			CsiDriver   interface{}            `json:"csiDriver"`
+		} `json:"nodes"`
+
+		RawResults struct {
+			Cnfcertificationtest struct {
+				Testsuites struct {
+					Testsuite struct {
+						Testcase []testCase `json:"testcase"`
+					} `json:"testsuite"`
+				} `json:"testsuites"`
+			} `json:"cnf-certification-test"`
+		} `json:"rawResults"`
+	} `json:"claim"`
+}
+type cnistruct []struct {
```

Comment:
Suggestion: I would change this to just `type Cni struct {` (without the "[]"). If you want to keep your name, use a camelCase `cniStruct` version, please.